dynamiclinks/app/src/main/java/com/google/firebase/quickstart/deeplinks/kotlin/MainComposeActivity.kt

```diff
@@ -0,0 +1,293 @@
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            DynamicLinksTheme {
+                Surface(
+                    modifier = Modifier.fillMaxSize(),
+                    color = MaterialTheme.colors.background
+                ) {
+                    MainAppView()
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun MainAppView(
+    lifecycleOwner: LifecycleOwner = LocalLifecycleOwner.current,
+    dynamicLinksViewModel: DynamicLinksViewModel = viewModel(factory = DynamicLinksViewModel.Factory)
+){
+    val context = LocalContext.current
+    val activity = context.findActivity()
+    val intent = activity?.intent
+    val uriPrefix = stringResource(R.string.dynamic_links_uri_prefix)
+    var newDeepLink by remember { mutableStateOf("") }
+    var openDialog by remember { mutableStateOf(false) }
+
+    DisposableEffect(lifecycleOwner) {
+        val observer = LifecycleEventObserver { _, event ->
+            // Configure Firebase Dynamic Links when the screen is created
+            if (event == Lifecycle.Event.ON_CREATE) {
+                // Validate that the developer has set the app code.
+                dynamicLinksViewModel.validateAppCode(uriPrefix)
+
+                newDeepLink = dynamicLinksViewModel.buildDeepLink(uriPrefix, Uri.parse(DEEP_LINK_URL), 0).toString()
+
+                intent?.let {
+                    dynamicLinksViewModel.getDynamicLink(it)
+                }
+            }
+        }
+
+        lifecycleOwner.lifecycle.addObserver(observer)
+
+        onDispose {
+            lifecycleOwner.lifecycle.removeObserver(observer)
+        }
+    }
+
+    val snackbarHostState = remember { SnackbarHostState() }
+    val channel = remember { Channel<Int>(Channel.Factory.CONFLATED) }
+    val scaffoldState = rememberScaffoldState(snackbarHostState = snackbarHostState)
+
+    // Checks if a valid uri has been updated, show only once
+    LaunchedEffect(key1 = true) {
+        dynamicLinksViewModel.validUriPrefix.collect { flag ->
+            if (!flag) {
+                openDialog = true
+            }
+        }
+    }
+
+    // Checks if a deep link is used to open app
+    LaunchedEffect(key1 = channel) {
+        dynamicLinksViewModel.deepLink.collect { deepLink ->
+            if (deepLink.isNotEmpty()) {
+                snackbarHostState.showSnackbar("Found deep link!")
+            }
+        }
+    }
+    Scaffold(
+        scaffoldState = scaffoldState,
+        topBar = {
+            TopAppBar(
+                backgroundColor = colorResource(R.color.colorPrimary)
+            ) {
+                Text(
+                    text = stringResource(R.string.app_name),
+                    style = MaterialTheme.typography.h6,
+                    textAlign = TextAlign.Center,
+                    modifier = Modifier.padding(8.dp),
+                    color = Color.White
+                )
+            }
+        },
+        content = { it
```

Comment:
This `it` seems like it's not being used here. Is it needed?
```suggestion
        content = {
```

Reply:
Yeah what I did here was just consume it without using it, because since Compose 1.2.0 it's required to use padding parameter to be passed into Scaffold content composable. In my next commit, what I did instead is to use a column to be able to use the paddingValue from `it`.